roles/precheck/tasks/pci_passthrough.yml

```diff
@@ -0,0 +1,93 @@
+---
+- name: "Run PCI passthrough pre-flight checks"
+  when: pci_passthrough_enabled | default(false)
+  block:
+    - name: "Gather additional hardware facts for CPU vendor and flags"
+      ansible.builtin.setup:
+        gather_subset: hardware
+
+    - name: "Assert CPU architecture is x86_64"
+      ansible.builtin.assert:
+        that: ansible_facts['architecture'] == 'x86_64'
+        fail_msg: |
+          Passthrough is currently only supported on x86_64 architecture.
+          Current architecture: {{ ansible_facts['architecture'] }}
+
+    - name: "Set virtualization flag facts based on CPU vendor"
+      ansible.builtin.set_fact:
```

Comment:
When using _asd variables the convention in one-deploy is that you should not use them as real facts, instead you could do something like:

```
- vars:
        _virtualization_flag: >-
          {{ 'vmx' if 'GenuineIntel' in ansible_facts['processor'] else
             'svm' if 'AuthenticAMD' in ansible_facts['processor'] else '' }}
        _slat_flag: >-
          {{ 'ept' if 'GenuineIntel' in ansible_facts['processor'] else
             'npt' if 'AuthenticAMD' in ansible_facts['processor'] else '' }}
  block: []
```
Then the code that uses "local" vars goes inside the block. If you need them as real facts to use them later try not to prepend them with _. :slightly_smiling_face: